### PR TITLE
Added DateTime GraphQL Scalar definition

### PR DESCRIPTION
**Pull request by Unraveler**
- Added new DateTime scalar definition for GraphQL
- Added setDateTimeConfig to Jackson Config objectMapper 

please review @terrestris/devs

### CODE REVIEW

File: shogun-lib/src/main/java/de/terrestris/shoguncore/config/JacksonConfig.java [hunk 1]
```diff
@@ -5,6 +5,7 @@
 import com.fasterxml.jackson.databind.MapperFeature;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.databind.util.ISO8601DateFormat;
```
Comment (dnlkoch) [inline shogun-lib/src/main/java/de/terrestris/shoguncore/config/JacksonConfig.java:8]: I think this line can be removed.

File: shogun-lib/src/main/java/de/terrestris/shoguncore/model/BaseEntity.java [hunk 2]
```diff
@@ -1,5 +1,6 @@
 package de.terrestris.shoguncore.model;
 
+import com.fasterxml.jackson.annotation.JsonFormat;
```
Comment (dnlkoch) [inline shogun-lib/src/main/java/de/terrestris/shoguncore/model/BaseEntity.java:3]: Can also be removed.

**Review by dnlkoch — APPROVED** [review]

File: shogun-lib/src/main/java/de/terrestris/shoguncore/graphql/scalar/DateTimeScalar.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            return parseDateFromAstLiteral(input);
+        }
+    });
+
+    private static Object serializeDate(Object dataFetcherResult) {
+        return objectMapper().convertValue(dataFetcherResult, String.class);
+    }
+
+    private static Object parseDateFromVariable(Object dataFetcherResult) {
+        return objectMapper().convertValue(dataFetcherResult, String.class);
```
Comment (simonseyock) [inline shogun-lib/src/main/java/de/terrestris/shoguncore/graphql/scalar/DateTimeScalar.java:41]: i think this should be like this:
```
if (input instanceof String) {
                return objectMapper.readValue((String) input, Date.class);
            }
            throw new CoercingParseValueException("...");
```

File: shogun-lib/src/main/java/de/terrestris/shoguncore/graphql/scalar/DateTimeScalar.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    private static Object parseDateFromAstLiteral(Object dataFetcherResult) {
+        return objectMapper().convertValue(dataFetcherResult, String.class);
```
Comment (simonseyock) [inline shogun-lib/src/main/java/de/terrestris/shoguncore/graphql/scalar/DateTimeScalar.java:45]: this should be like this:
```
if (input instanceof StringValue) {
    return objectMapper.readValue(((StringValue) input).getValue(), Date.class);
}
throw new CoercingParseValueException("...");
```

**Pull request merged**